### PR TITLE
Test if circle first before angular corrections

### PR DESCRIPTION
**Pull request by ghost**
### Description

Test if circle first before angular corrections. 
Also, if it is a circle, then don't perform angular corrections because
the correct state is already now known. Also, if circle, respect user's
wish for either clockwise or counter-clockwise motion instead of forcing
CCW motion. Lastly, test for perfect zero case to prevent scenario where
perfect zero CW is -2pi, but perfect zero CCW is left zero. It just
makes sense to handle this where the correct states are already being
set rather than in the switch statement which doesn't apply it
consistently across 0 CW vs 0 CCW.

Basically:
Rearranged code to be more logical and reduce unnecessary computation in cases where a perfect circle was required. Also ensured that logical operations are applied consistently across G2 and G3 situations where angular travel is near or at zero. 

### Benefits

1.) reduces unnecessary computation for circles. 
2.) perfect consistency across equivalent circumstances in opposite directions

### Configurations

default config files work perfectly 
Arc Support required

### Related Issues

#20258  
#20292

### CODE REVIEW

**Comment by sjasonsmith** [comment]
> //even though a circle is a circle we should still probably respect users wish on direction of motion via G2/G3

This might not matter for printing, but is very important for milling. Switching CCW/CW is swithing between climb-milling and conventional-milling, which have very different cutting characteristics.

**Comment by ghost** [comment]
oh perfect. in that case this is a must then.

**Comment by yysh12** [comment]
oh wait give me sec. looks like I messed something up.

**Comment by ghost** [comment]
If one wants to suggest additional changes to an existing PR, is it possible to do that without opening a new PR? Or is it that once a merge is accepted the PR is closed and cannot accept additional changes?

**Comment by matbhe** [comment]
Hello

I've done some G2/G3 tests. The following G2 - Test failed (no movement):
G21
G90
G0 X-3.50 Y0.00
G2 I3.50 J0

Regards Matthias

**Comment by sjasonsmith** [comment]
@matbhe please report a new issue for problems you encounter. That ensures they don't get lost as comments on closed PRs are likely to.
